python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -29,6 +29,18 @@
 
 logger = Ganga.Utility.logging.getLogger(modulename=1)
 
+_imported_GangaList = None
+
+do_not_copy = ['_proxyObject', '_proxyClass', '_data', '_index_cache', '_parent', '_registry', '_impl']
+
+def _getGangaList():
+    global _imported_GangaList
+    if _imported_GangaList is None:
+        from Ganga.GPIDev.Lib.GangaList.GangaList import GangaList
+        _imported_GangaList = GangaList
+    return _imported_GangaList
+
+
```

Comment:
Yes this is what's going on.
If not this way can you suggest a better alternative to avoid the circular importing?
I'm all for a better solution as I'm constantly running into this in Python
I have tried placing the equivalent code in places where GangaList is called but end up with some of the code being called in the case of constructing GangaList so ended up duplicating this several times (hence the helper function)